packages/pigeon/lib/cpp_generator.dart

```diff
@@ -36,13 +38,17 @@ class CppOptions {
   /// A copyright header that will get prepended to generated code.
   final Iterable<String>? copyrightHeader;
 
+  /// The path to the output header file location.
+  final String? cppHeaderOut;
```

Comment:
Having this and `header` is confusing. How about `headerOut` (the `cpp` is redundant with the options structure name) for this, and `headerIncludePath` for what's currently called `header`?